lib/screenplay/suppressed_predictions.ex

```diff
@@ -37,6 +40,73 @@ defmodule Screenplay.SuppressedPredictions do
     |> Repo.all()
   end
 
+  @doc """
+  Get all suppressed predictions modified for transit data
+  """
+  @spec get_all_suppressed_predictions_for_data() :: [
+          %{
+            route_id: String.t(),
+            stop_id: String.t(),
+            direction_id: integer(),
+            suppression_type: :terminal | :stop | :none
+          }
+        ]
+  def get_all_suppressed_predictions_for_data do
+    suppressed_predictions = get_all_suppressed_predictions()
+
+    Places.get()
+    |> Enum.flat_map(fn
+      %Screenplay.Places.Place{id: "place-jfk"} ->
+        Enum.map(
+          PredictionSuppressionUtils.jfk_umass_child_stop_data(),
+          fn %{
+               route_id: route_id,
+               stop_id: child_stop_id,
+               direction_id: direction_id
+             } ->
+            %{
+              stop_id: child_stop_id,
+              route_id: route_id,
+              direction_id: direction_id,
+              suppression_type:
+                PredictionSuppressionUtils.suppression_type(
+                  suppressed_predictions,
+                  child_stop_id
+                )
+            }
+          end
+        )
+
+      %Screenplay.Places.Place{id: place_stop_id, screens: screens} ->
+        Enum.flat_map(screens, fn
+          %Screenplay.Places.Place.PaEssScreen{routes: routes} ->
+            routes
+            |> Enum.uniq_by(fn route -> {route.id, route.direction_id} end)
+            |> Enum.filter(&PredictionSuppressionUtils.valid_route?(&1.id))
+            |> Enum.map(fn route ->
```

Comment:
General note: Check out the special `for` syntax, which can be an elegant way of expressing chains of `flat_map`, `map`, `filter`, and `uniq`.